pdig.cc

```diff
@@ -122,34 +82,42 @@ void handle_syscall(pid_t pid, pdig_process_context& pctx, bool enter)
 {
 	struct user_regs_struct regs;
 	uint64_t context[CTX_SIZE];
+	unsigned long syscall_nr;
 
 	if (ptrace(PTRACE_GETREGS, pid, &regs, &regs) < 0) {
 		WARN("PTRACE_GETREGS failed for pid %d", pid);
 		return;
 	}
 	fill_context(context, pid, pctx.pid, regs);
+	syscall_nr = context[CTX_SYSCALL_ID];
 
 	DEBUG("pid=%d tid=%d syscall=%ld ret=%ld enter=%d\n",
-		pctx.pid, pid, context[CTX_SYSCALL_ID], context[CTX_RETVAL], enter);
+		pctx.pid, pid, syscall_nr, context[CTX_RETVAL], enter);
 
-	if(context[CTX_SYSCALL_ID] == __NR_rt_sigreturn)
+	if(syscall_nr == __NR_rt_sigreturn)
 	{
 		DEBUG("rt_sigreturn, faking exit event\n");
 		on_syscall(context, false);
 	}
-	else if(context[CTX_SYSCALL_ID] == __NR_execve && context[CTX_RETVAL] == 0)
+	else if(syscall_nr == __NR_execve && context[CTX_RETVAL] == 0)
```

Comment:
does execveat fall into this case too?

Reply:
Possibly. I'll need to test it. https://github.com/falcosecurity/pdig/issues/14